src/launcher.sh

```diff
@@ -15,64 +19,25 @@ if [ "$1" = "--shell" ]; then
   exit $?
 fi
 
-# Quote all strings for insertion into json
-# busybox does not implement ${VAR@Q} substitution to quote variables
+if [[ "${CONTAINER_PRESERVE_CONFIG:-}" == "true" ]]; then
+  log_warn "CONTAINER_PRESERVE_CONFIG set to true."
+  log_warn "options.json and admin.txt will not be modified."
+else
+  # Update configuration file
+  mkdir -p /data/Config >& /dev/null
+  log "Generating options.json file."
+  ./set_options.js > /data/Config/options.json
 
-if [[ "${FOUNDRY_AWS_CONFIG:-}" ]]; then
-  if [[ $FOUNDRY_AWS_CONFIG == "true" ]];then
-    FOUNDRY_AWS_CONFIG=true
+  # Save admin access key to file if set.  Delete file if unset.
+  if [[ "${FOUNDRY_ADMIN_KEY:-}" ]]; then
+    log "Setting 'Admin Access Key'."
+    echo "${FOUNDRY_ADMIN_KEY}" | ./set_password.js > /data/Config/admin.txt
   else
-    FOUNDRY_AWS_CONFIG=\"${FOUNDRY_AWS_CONFIG}\"
+    log_warn "No 'Admin Access Key' has been configured."
+    rm /data/Config/admin.txt >& /dev/null || true
```

Comment:
This rm will remove a pre-generated `admin.txt` file if the admin key is not specified. I'm not 100% sure all users would want that. May want to `log_warn` and let a user know if the key is empty and the file already exists.

Reply:
This isn't new functionality.   It just got bumped around a bit.  

This behavior is well-documented in the README: 
See: `FOUNDRY_ADMIN_KEY` of https://github.com/felddy/foundryvtt-docker/blob/develop/README.md#optional